[PT 3] State management and cart page

KauhareniaNastassia opened the pull request:
Add cart page
Add widgets for cart state
Add bloc, state, event, models for cart page
Add empty cart page

https://github.com/KauhareniaNastassia/food_delivery_app/assets/104850152/ed0213a0-39a8-4c1b-8034-3c5d886e31fc

Commit:
[PT-2] Theme and navigation
Add some refactoring

Commit:
[PT-3] State management and cart page
Add cart page
Add widgets for cart state
Add bloc, state, event, models for cart page
Add empty cart page

Commit:
[PT-3] State management and cart page
Add cart page
Add widgets for cart state
Add bloc, state, event, models for cart page
Add empty cart page

Commit:
[PT-3] State management and cart page
Add some refactoring

Commit:
[PT-3] State management and cart page
Add some refactoring

Commit:
[PT-3] State management and cart page
Add some refactoring

AliaksandrHuzich commented on core_ui/lib/src/widgets/buttons/primary_button.dart:
```diff
@@ -1,53 +1,32 @@
+          primary: AppColors.primaryColor,
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(15)),
+          )),
```
,

AliaksandrHuzich commented on features/main_page_view/lib/src/bloc/main_page/bloc.dart:
```diff
@@ -15,15 +14,17 @@ class MenuBloc extends Bloc<MenuEvent, MenuState> {
     try {
-      final List<MenuItemModel> menu = await _fetchMenuItemsUseCase.execute(const NoParams());
+      final List<MenuItemModel> menu =
+          await _fetchMenuItemsUseCase.execute(const NoParams());
```
,

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item.dart:
```diff
@@ -1,97 +1,139 @@
+}
+
+class _MenuItemState extends State<MenuItem> {
+  late bool isItemInCart;
```
remove these

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item.dart:
```diff
@@ -1,97 +1,139 @@
+class _MenuItemState extends State<MenuItem> {
+  late bool isItemInCart;
+
+  _isItemInShoppingCart() {
```
костыль

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -1,170 +1,211 @@
+class _MenuItemDetailsScreenState extends State<MenuItemDetailsScreen> {
+  late bool isItemInCart;
+
+  _isItemInShoppingCart() {
```
same

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -1,170 +1,211 @@
+              SliverAppBar(
+                leading: IconButton(
+                  onPressed: () {
+                    Navigator.pop(context);
```
to bloc

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -1,170 +1,211 @@
+                          ? Image.network(
+                              widget.menuItem.image,
+                              fit: BoxFit.cover,
+                              errorBuilder: (context, error, stackTrace) =>
```
_ __ ___

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -1,170 +1,211 @@
+                            topRight: Radius.circular(50),
+                          ),
+                          border: Border.all(
+                            width: 0,
```
?

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/menu_item_details_screen.dart:
```diff
@@ -1,170 +1,211 @@
+import 'package:core/core.dart';
```
раздели на отдельные файлы

AliaksandrHuzich commented on features/main_page_view/lib/src/ui/components/widgets/add_to_cart_button.dart:
```diff
@@ -0,0 +1,36 @@
+              isItemInCart ? AppColors.secondaryColor : AppColors.primaryColor,
+          shape: const RoundedRectangleBorder(
+            borderRadius: BorderRadius.all(Radius.circular(15)),
+          )),
```
,

AliaksandrHuzich commented on features/shopping_cart/lib/src/bloc/bloc.dart:
```diff
@@ -0,0 +1,116 @@
+        if (shoppingCartItems[i].menuItem == event.menuItem) {
+          shoppingCartItems[i].amount++;
+          itemIsInCart = true;
+          break;
```
break

AliaksandrHuzich replied:
while

AliaksandrHuzich commented on features/shopping_cart/lib/src/bloc/state.dart:
```diff
@@ -0,0 +1,27 @@
+  final ShoppingCartModel shoppingCart;
+
+  const ShoppingCartState({
+    this.shoppingCart = const ShoppingCartModel(
```
именнованый конструктор ShoppingCartModel.empty

AliaksandrHuzich commented on features/shopping_cart/lib/src/ui/empty_shopping_cart_screen.dart:
```diff
@@ -0,0 +1,43 @@
+
+  @override
+  Widget build(BuildContext context) {
+    final size = MediaQuery.of(context).size;
```
types

AliaksandrHuzich commented on features/shopping_cart/lib/src/ui/empty_shopping_cart_screen.dart:
```diff
@@ -0,0 +1,43 @@
+          PrimaryButton(
+            buttonTitle: 'Go to menu',
+            onPressed: () {
+              context.navigateTo(
```
onTap

AliaksandrHuzich commented on features/shopping_cart/lib/src/ui/widgets/item_amount.dart:
```diff
@@ -0,0 +1,60 @@
+            shoppingCartBloc.add(
+              (AddShoppingCartItemEvent(
+                menuItem: shoppingCartItem,
+              )),
```
,

Commit:
[PT-3] State management and cart page
Correct some issues
Add small components for menu item details page and for menu item page
Add bloc for navigation between pages

Commit:
[PT-3] State management and cart page
Add some refactoring
